cheatsheets_excluded/Content_Security_Policy_Cheat_Sheet.md

```diff
@@ -1,203 +1,199 @@
 
-Specifications of the CSP standard can be found the following locations:
+CSP can be delivered to the user agent in different techniques.
+1. `Content-Security-Policy` HTTP response header field. This is the most preferred technique.
+1. `<meta>` HTML element with `http-equiv` attribute set to `Content-Security-Policy`. These elements need to be placed as early as possible in the documents.
+1. `Content-Security-Policy-Report-Only` HTTP response header field. This header is used when the developer is unsure of the CSP behavior and wants to monitor it, instead of enforcing it.
 
--   Latest Revision - <https://w3c.github.io/webappsec/specs/content-security-policy/>
--   Latest Version (CSP2) - <http://www.w3.org/TR/CSP2/>
--   CSP 1.0 - <http://www.w3.org/TR/2012/CR-CSP-20121115/>
+## HTTP Headers
 
-CSP Basics
-==========
-
-CSP consists of a series of directives. CSP has also evolved over two major revisions. Most browsers support 1.0, and adoption of CSP2 has been incremental.
+The following are headers for CSP.
 
-HTTP Headers
-------------
+- `Content-Security-Policy` : W3C Spec standard header. Supported by Firefox 23+, Chrome 25+ and Opera 19+
+- `Content-Security-Policy-Report-Only` : W3C Spec standard header. Supported by Firefox 23+, Chrome 25+ and Opera 19+, whereby the policy is non-blocking ("fail open") and a report is sent to the URL designated by the `report-uri` directive. This is often used as a precursor to utilizing CSP in blocking mode ("fail closed")
+- `DO NOT` use X-Content-Security-Policy or X-WebKit-CSP. Their implementations are obsolete (since Firefox 23, Chrome 25), limited, inconsistent, and incredibly buggy.
 
-The following are headers for CSP.
+# CSP Directives
 
--   **Content-Security-Policy** : W3C Spec standard header. Supported by Firefox 23+, Chrome 25+ and Opera 19+
--   **Content-Security-Policy-Report-Only** : W3C Spec standard header. Supported by Firefox 23+, Chrome 25+ and Opera 19+, whereby the policy is non-blocking ("fail open") and a report is sent to the URL designated by the **report-uri** directive. This is often used as a precursor to utilizing CSP in blocking mode ("fail closed")
--   **DO NOT** use X-Content-Security-Policy or X-WebKit-CSP. Their implementations are obsolete (since Firefox 23, Chrome 25), limited, inconsistent, and incredibly buggy.
+Multiple types of directives exist that allow the developer to granularly control the flow of the policies.
 
-Directives
-----------
+## Fetch Directives
 
-The following is a listing of directives, and a brief description.
+Fetch directives tell the browser the locations to trust and load resources from.
 
-### CSP 1.0 Spec
+- `default-src` is a fallback directive for the other fetch directives. Directives that are specified have no inheritance, yet directives that are not specified will fall back to the value of `default-src`.
+- `child-src` allows the developer to control nested browsing contexts and worker execution contexts.
```

Comment:
Add this remark from Mozilla about this directive:

*Instead of child-src, authors who wish to regulate nested browsing contexts and workers should use the frame-src and worker-src directives, respectively.*